src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -245,7 +250,18 @@ object RefArrayUtils {
       Array.copy(array, 0, newArray, 0, end)
       newArray
     }
-    if (n <= arrayLength) array else growArray
+    def shrinkArray = {
+      var newSize: Long = math.max(arrayLength * 2, 8)
```

Comment:
might be worth `8` being a constant (MinArrayLength) or similar.
Not sure why you are doing `arrayLength * 2` is it meant to be `arrayLength / 2`?